bob-common/src/mapper.rs

```diff
@@ -132,8 +133,21 @@ impl Virtual {
                     None
                 }
             })
-            .take(count)
-            .collect()
+            .collect();
+        let mut result = vec![];
+        for &node in nodes.iter() {
+            if result.len() < count && node.connection_available() {
```

Comment:
This heavily overlaps with https://github.com/qoollo/bob/pull/629. Once #629 will be merged, this PR should be updated